modules/nw-cluster-mtu-change.adoc

```diff
@@ -0,0 +1,309 @@
+// Module included in the following assemblies:
+//
+// * networking/configuring-cluster-network-mtu.adoc
+
+[id="nw-cluster-mtu-change_{context}"]
+= Changing the cluster MTU
+
+As a cluster administrator, you can change the maximum transmission unit (MTU) for your cluster. The migration is disruptive and nodes in your cluster might be temporarily unavailable as the MTU update rolls out.
+
+.Prerequisites
+
+* You installed the OpenShift CLI (`oc`).
+* You are logged in to the cluster with a user with `cluster-admin` privileges.
+* You identified the target MTU for your cluster. The correct MTU varies depending on the cluster network provider that your cluster uses:
+** *OVN-Kubernetes*: The cluster MTU must be set to `100` less than the lowest MTU value in your cluster.
+** *OpenShift SDN*: The cluster MTU must be set to `50` less than the lowest MTU value in your cluster.
+
+.Procedure
+
+To increase or decrease the MTU for the cluster network complete the following procedure.
+
+. To obtain the current MTU for the cluster network, enter the following command:
++
+[source,terminal]
+----
+$ oc describe network.config cluster
+----
+
+. Get the NetworkManager connection name:
+
+.. To find the primary network interface, enter the following command:
++
+[source,terminal]
+----
+$ oc debug node/<node_name> -- chroot /host ip route list match 0.0.0.0/0 | awk '{print $5 }'
+----
++
+--
+where:
+
+`<node_name>`:: Specifies the name of a node in your cluster.
+--
+
+.. To find the connection that NetworkManager created for the interface name returned from the previous command, enter the following command:
++
+[source,terminal]
+----
+$ oc debug node/<node_name> -- chroot /host nmcli c | grep $(ip route list match 0.0.0.0/0 | awk '{print $5 }')
+----
++
+.Example output
+[source,text]
+----
+Wired connection 1  46da4a6a-xxxx-xxxx-xxxx-ac0ca900f213  ethernet  ens3
+----
+
+. Prepare your configuration for the hardware MTU:
+
+** If your hardware MTU is specified in a NetworkManager connection configuration, complete the following steps. This approach is the default for {product-title} if you do not explicitly specify your network configuration with DHCP, a kernel command line, or some other method.
+
+... To create a new NetworkManager connection configuration with the hardware MTU, enter the following command:
++
+[source,terminal]
+----
+$ oc debug node/<node_name> -- chroot /host cat /etc/sysconfig/network-scripts/ifcfg-<connection_name> > ifcfg.txt
```

Comment:
This is a bit disjoint to the previous step where we were assuming a NM profile vs here were we are assuming an ifcfg file. They are not the same thing. I don't know which is the default, but I guess the most frequent case would be an auto-generated NM profile.
My guess is that it would be better to doc here how to extract the NM profile obtained on the previous step  and set the MTU there. And then have alternate instructions in case the config is through an ifcfg file if required.

Reply:
@jcaamano, oh, I pulled this from an unpublished KCS; I don't actually have any idea how to do this, so this was my best guess. Specifically: https://access.redhat.com/solutions/5307301

Reply:
So for an NM connection profile, taking the UUID from the previous example....

```
# get the profile path on the node
profile="$(oc debug node/<node_name> -- chroot /host nmcli -g UUID,FILENAME c show | grep 46da4a6a-xxxx-xxxx-xxxx-ac0ca900f213 | cut -d: -f2)"

# extract the profile
oc debug node/<node_name> -- chroot /host cat "$profile" > "$(basename $profile)"

# edit profile
# set 802-3-ethernet.mtu for MTU
# set connection.autoconnect-priority to increase priority over other existing profiles for the same device if required
# remove connection.uuid
# change connection.id to rename the connection as required
vi "$(basename $file)"
```

Then you would follow on with the MacineConfig part to write the profile in `/etc/NetworkManager/system-connections/`

Hopefully we could refer to RHEL8 docs i.e.
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_and_managing_networking/assembly_manually-creating-networkmanager-profiles-in-key-file-format_configuring-and-managing-networking